vmm/src/vm.rs

```diff
@@ -768,7 +768,32 @@ impl Vm {
         )
         .map_err(|_| Error::CmdLine)?;
         let vcpu_count = self.config.cpus.cpu_count;
-        let end_of_range = GuestAddress((1 << get_host_cpu_phys_bits()) - 1);
+
+        #[cfg(feature = "acpi")]
+        let rsdp_addr = {
+            let end_of_range = GuestAddress((1 << get_host_cpu_phys_bits()) - 1);
+
+            let mem_end = mem.end_addr();
+            let start_of_device_area = if mem_end < arch::layout::MEM_32BIT_RESERVED_START {
+                arch::layout::RAM_64BIT_START
+            } else {
+                mem_end.unchecked_add(1)
+            };
+
+            use crate::config::ConsoleOutputMode;
+            crate::acpi::create_acpi_tables(
+                &mem,
+                vcpu_count,
+                self.config.serial.mode != ConsoleOutputMode::Off,
+                start_of_device_area,
+                end_of_range,
+                self.devices.virt_iommu(),
+            )
+        };
+
+        #[cfg(not(feature = "acpi"))]
+        let rsdp_addr = GuestAddress(0);
```

Comment:
This is the cleanest I could come up with.

Reply:
What about:

```rust
        #[allow(unused_mut, unused_assignments)]
        let mut rsdp_addr: Option<GuestAddress> = None;

        #[cfg(feature = "acpi")]
        {
            rsdp_addr = {
                let end_of_range = GuestAddress((1 << get_host_cpu_phys_bits()) - 1);

                let mem_end = mem.end_addr();
                let start_of_device_area = if mem_end < arch::layout::MEM_32BIT_RESERVED_START {
                    arch::layout::RAM_64BIT_START
                } else {
                    mem_end.unchecked_add(1)
                };

                use crate::config::ConsoleOutputMode;
                crate::acpi::create_acpi_tables(
                    &mem,
                    vcpu_count,
                    self.config.serial.mode != ConsoleOutputMode::Off,
                    start_of_device_area,
                    end_of_range,
                    self.devices.virt_iommu(),
                )
            };
        }
```

Reply:
What about `GuestAddress(0)` instead of the `Option` ?

Reply:
I did it as you proposed.